Sources/SwiftGRPC/Core/Channel.swift

```diff
@@ -89,17 +87,17 @@ public class Channel {
     let argumentWrappers = arguments.map { $0.toCArg() }
 
     underlyingChannel = withExtendedLifetime(argumentWrappers) {
-        var argumentValues = argumentWrappers.map { $0.wrapped }
-        return cgrpc_channel_create_secure(address, certificates, clientCertificates, clientKey, &argumentValues, Int32(arguments.count))
+      var argumentValues = argumentWrappers.map { $0.wrapped }
+      return cgrpc_channel_create_secure(address, certificates, clientCertificates, clientKey, &argumentValues, Int32(arguments.count))
     }
     completionQueue = CompletionQueue(underlyingCompletionQueue: cgrpc_channel_completion_queue(underlyingChannel), name: "Client")
     completionQueue.run() // start a loop that watches the channel's completion queue
   }
 
   deinit {
-    connectivityObservers.forEach { $0.shutdown() }
-    cgrpc_channel_destroy(underlyingChannel)
-    completionQueue.shutdown()
+    self.connectivityObserver?.shutdown()
```

Comment:
Wrap this in a `mutex.synchronize`? I guess it's unlikely that some other code is accessing `connectivityObserver` while we are already in `deinit`, but better safe than sorry?

Reply:
Yep, I have that change locally in another branch - meant to push it here too